apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -65,22 +76,27 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
                   )
                   .map(([name]) => name);
 
-    useEffect(() => {
-        const newSearchParams = new URLSearchParams(searchParams);
-        newSearchParams.set('module', selectedModule);
-        setSearchParams(newSearchParams);
-    }, [selectedModule, setSearchParams, searchParams]);
-
     const submitSearch = useCallback(() => {
-        setSelectedModule((prev: string) =>
-            filteredModules.length === 1 ? filteredModules[0] : prev
-        );
-    }, [filteredModules]);
+        if (filteredModules.length === 1) {
+            const convertedSearchParams = new URLSearchParams(searchParams);
+            convertedSearchParams.set('module', filteredModules[0]);
+            setSearchParams(convertedSearchParams);
+        }
+    }, [filteredModules, setSearchParams, searchParams]);
+
+    const onChangeModule = useCallback(
+        (newModule: string) => () => {
```

Comment:
I don't think there's a lot of value in this double callback approach. I'd just remove the nested callback then do `onClick={() => onChangeModule(name)}`.

Reply:
Removing the useCallback causes the user to no longer be able to switch module.